app/src/main/java/io/pslab/activity/DataLoggerActivity.java

```diff
@@ -94,7 +118,127 @@ public boolean onOptionsItemSelected(MenuItem item) {
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.activity_data_logger_menu, menu);
+        return true;
+    }
+
+    private void importLog() {
+        AlertDialog alertDialog;
+        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(this);
+        LayoutInflater inflater = this.getLayoutInflater();
+        final View dialogView = inflater.inflate(R.layout.import_log_device_type_alert_layout, null);
+        dialogBuilder.setView(dialogView);
+        dialogBuilder.setPositiveButton("Select", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                RadioGroup votingAlgoRadioGroup = dialogView.findViewById(R.id.import_log_device_radio_group);
+                try{
+                    RadioButton selectedRadioButton = dialogView.findViewById(votingAlgoRadioGroup.getCheckedRadioButtonId());
+                    selectedDevice = selectedRadioButton.getText().toString();
+                    selectFile();
+                }catch (Exception e){
+                    Toast.makeText(DataLoggerActivity.this, getResources().getString(R.string.import_data_log_no_selection_error), Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+        alertDialog = dialogBuilder.create();
+        alertDialog.show();
+    }
+
+    private void selectFile() {
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.setType("*/*");
+        startActivityForResult(intent, 100);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        if (requestCode == 100) {
+            Uri uri = data.getData();
+            String path = uri.getPath();
+            path = path.replace("/root_path/", "/");
+            File file = new File(path);
+            getFileData(file);
+        }
+    }
+
+    private void getFileData(File file) {
+        if (selectedDevice != null && selectedDevice.equals(getResources().getString(R.string.baro_meter))) {
+            try {
+                FileInputStream is = new FileInputStream(file);
+                BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+                String line = reader.readLine();
+                int i = 0;
+                long block = 0, time = 0;
+                while(line != null){
+                    if (i != 0) {
+                        String[] data = line.split(",");
+                        try {
+                            time += 1000;
+                            BaroData baroData = new BaroData(time, block, Float.valueOf(data[2]), Double.valueOf(data[3]), Double.valueOf(data[4]));
+                            realm.beginTransaction();
+                            realm.copyToRealm(baroData);
+                            realm.commitTransaction();
+                        }catch (Exception e) {
```

Comment:
@neel1998 Have you reformatted the code, I don't think so, it shouldn't be this way!

Reply:
@abhinavraj23  can you please explain what do you mean by "reformatted"?

Reply:
@neel1998 In Android Studio you can automatically format your code. Please follow this link https://stackoverflow.com/questions/16580171/code-formatting-shortcut-in-android-studio

Reply:
@abhinavraj23  Oh okay, thanks. I will make the changes

Reply:
@neel1998 What I meant to say was there should be a space between the bracket and the "catch" word, in Android studio you can automatically do these minor formatting!

Reply:
@abhinavraj23  great. i will make the necessary changes.